docs/py_docs.rst

```diff
@@ -1097,28 +1179,43 @@ For example:
       Requirements Document, LPM-17, URL https://ls.st/LPM-17
    """
 
+Attributes set in \_\_init\_\_ methods
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
 In many classes, public attributes are set in the ``__init__`` method.
 The best way to document these public attributes is by declaring the attribute at the class level and including a docstring with that declaration:
 
 .. code-block:: python
+   :emphasize-lines: 12-18
 
    class Metric(object):
        """Verification metric.
 
-       [...]
+       Parameters
+       ----------
+       name : `str`
+           Name of the metric.
+       unit : `astropy.units.Unit`
+           Units of the metric.
        """
 
        name = None
-       """Name of the metric (`str`)."""
+       """Name of the metric (`str`).
+       """
 
        unit = None
-       """Units of the metric (`astropy.units.Unit`)."""
+       """Units of the metric (`astropy.units.Unit`).
+       """
 
-       def __init__(name, unit):
+       def __init__(self, name, unit):
            self.name = name
            self.unit = unit
 
-Private attributes (prefixed by underscores: ``self._myAttribute``) do not need to be documented with docstrings.
+Notice that the :ref:`parameters <py-docstring-parameters>` to the ``__init__`` method are documented separately from the class attributes (highlighted).
+
+.. note::
+
+   Private attributes (prefixed by underscores: ``self._myAttribute``) do not need to be documented with docstrings.
```

Comment:
How do developers of the class (as opposed to users of the class) know what they are for then? If you document it with a docstring I thought it was hidden by default but you could use a flag to turn it on. I think it is good practice to document private methods and private attributes and if we don't give guidance here we'll have a mix of some with docstrings, some with commented text and some completely undocumented.